app/models/rating_vote.rb

```diff
@@ -22,7 +22,7 @@ def assign_article_rating
   end
 
   def permissions
-    return unless context == "explicit" && !user&.trusted && user_id != article&.user_id
+    return unless context == "explicit" && !user&.trusted? && user_id != article&.user_id
```

Comment:
unrelated to the change here - this `&& !user&.trusted? &&` check feels hard to understand

I had to spend time wondering why `user` would not be set (presumably either user_id is null or there's no associated record), what it means when user doesn't exist, whether this `unless _ && !(user && user.trusted?) && _` makes sense to me, and when it could possibly be clearer.

I'm probably just overreacting to negating a nil in a boolean context, or negating the predicate response, and wondering why those should be equivalent. It's also late, I might be tired.

Reply:
Oh boy is this hard to understand.  I would love to see this factored towards individual guard clauses.

Reply:
it looks like the nil cases are part of the builtin validations (when we're asserting the model association details).

However, I turned an `unless !x && !y && !z` to `if x || y || z` so I feel like I added something useful here.